praw/models/reddit/removal_reasons.py

```diff
@@ -32,25 +33,17 @@ class RemovalReason(RedditBase):
 
     STR_FIELD = "id"
 
-    def __eq__(self, other: Union[str, _RemovalReason]) -> bool:
-        """Return whether the other instance equals the current."""
-        if isinstance(other, str):
-            return other == str(self)
-        return isinstance(other, self.__class__) and str(self) == str(other)
-
-    def __hash__(self) -> int:
-        """Return the hash of the current instance."""
-        return hash(self.__class__.__name__) ^ hash(str(self))
-
     def __init__(
         self,
         reddit: Reddit,
         subreddit: Subreddit,
-        reason_id: str,
```

Comment:
Let's not rename this parameter, because it's part of the public interface of this class.

Reply:
Deprecating it.